config/v1/types_console.go

```diff
@@ -26,6 +28,31 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// route contains hostname and serving certificate for replacing the default route for the console.
+	// If a custom route is specified, a new route will be created with the provided hostname.
+	// The default console route will be maintained to reserve the default hostname for console if the custom route is removed.
+	// If not specified, default route will be 	used.
+	// +optional
+	Route ConsoleConfigRoute `json:"route,omitempty"`
```

Comment:
not in this group.  Put it on  your operator resource.

Reply:
is there a reason why it can live on the console config ?

Reply:
it's more that it shouldn't.  this is config that's only consumed by your operator, it's not global config that's consumed by others.  (the console operator publishes the public route hostname back to this resource under status, which *is* global information that's consumed by others, or can be)

Reply:
Our logic has been `a custom route is a normal thing a cluster-admin may configure`, thus it should exist on the top-level config.

That said, we can move it.

Reply:
How "normal" it is is not the right deciding factor for where it should go, the things l listed above are